src/NuGet.Core/NuGet.ProjectModel/PackageSpecWriter.cs

```diff
@@ -526,23 +532,38 @@ private static void SetFrameworkReferences(IObjectWriter writer, ISet<FrameworkD
             }
         }
 
-        private static void SetCentralDependencies(IObjectWriter writer, ICollection<CentralPackageVersion> centralPackageVersions)
+        private static void SetCentralDependencies(IObjectWriter writer, ICollection<CentralPackageVersion> centralPackageVersions, bool compressed)
         {
             if (!centralPackageVersions.Any())
             {
                 return;
             }
 
-            writer.WriteObjectStart("centralPackageVersions");
+            if (compressed)
+            {
+                SetValue(writer, "centralPackageVersionsHash", GetHash(centralPackageVersions).ToString());
```

Comment:
Do we have a "packageReferecesHash" in the PackageSpec?  If not, why is central package versions hash being special cased?  If it's a perf win for CPVM, then it should also be a perf win for PackageReference, PackageDownload, FrameworkReference as well, right?

Reply:
I think we can make the noop hash faster by not relying on generating a string representation.

Reply:
I did some investigations that generate an actual hash instead of using the PackageSpecWriter, and that seems faster in perf view, but it doesn't translate to overall improvements (ie noise from process start-up doesn't help). 

The problem is that the package spec gets mutated during restore in some situations, so my investigation is not complete.

Reply:
We discussed this offline. I removed the new public api. We can take this fix now and work on a different fix in a new PR.

Reply:
Personally, I'm ok with this small optimization for CPVM. I expect https://github.com/NuGet/Home/issues/10068 to improve things CPVM or not :)